src/main.cpp

```diff
@@ -0,0 +1,235 @@
+                }
+            }
+            else {
+                // GetObject
+                std::cout << "getobject detected" << std::endl;
+                co_await irods::s3::actions::handle_getobject(socket, parser, url);
+            }
+            break;
+        case boost::beast::http::verb::put:
+            if (parser.get().find("x-amz-copy-source") != parser.get().end()) {
+                // copyobject
+                std::cout << "Copyobject detected" << std::endl;
+                co_await irods::s3::actions::handle_copyobject(socket, parser, url);
+            }
+            else {
+                // putobject
+                std::cout << "putobject detected" << std::endl;
+                co_await irods::s3::actions::handle_putobject(socket, parser, url);
+                co_return;
+            }
+            break;
+        case boost::beast::http::verb::delete_:
+            if (segments.empty()) {
+                std::cout << "Deletebucket detected?" << std::endl;
+            }
+            else {
+                std::cout << "Deleteobject detected" << std::endl;
+                co_await irods::s3::actions::handle_deleteobject(socket, parser, url);
+                co_return;
+            }
+            break;
+        case boost::beast::http::verb::head:
+            // Probably just headbucket and headobject here.
+            // and headbucket isn't on the immediate list
+            // of endpoints.
+            if (url.segments().empty()) {
+                std::cout << "Headbucket detected" << std::endl;
+            }
+            else {
+                std::cout << "Headobject detected" << std::endl;
+                co_await irods::s3::actions::handle_headobject(socket, parser, url);
+                co_return;
+            }
+            break;
+            /*
+        case boost::beast::http::verb::delete_:
+            // DeleteObject
+            std::cout << "Deleteobject detected" << std::endl;
+            break;
+            */
+        default:
+            std::cerr << "Oh no..." << std::endl;
+            exit(37);
+            break;
+    }
+    char buf[512];
+    while (!parser.is_done()) {
+        std::cout << "Reading" << std::endl;
+        parser.get().body().data = buf;
+        parser.get().body().size = sizeof(buf);
+        std::cout << std::string_view((char*) parser.get().body().data, parser.get().body().size) << std::endl;
+        // Using async_read here causes it to completely eat the entire input without handling it properly.
+        auto read = co_await beast::http::async_read_some(socket, buffer, parser, asio::use_awaitable);
+        std::cout << "Read " << read << " bytes" << std::endl;
+        if (ec == beast::http::error::need_buffer) {
+            ec = {};
+        }
+        else if (ec) {
+            co_return;
+        }
+    }
+    co_return;
+}
+
+unsigned short port = 8080;
```

Comment:
Is this used anywhere other than in the function below?

I see a couple approaches here:
- Move it into the function where it's used
- Pass it as a function argument